ios/A0Auth0.m

```diff
@@ -101,31 +116,33 @@ - (NSString *)sign:(NSString*)value {
 
 - (NSDictionary *)generateOAuthParameters {
     NSString *verifier = [self randomValue];
-    NSString *bundleIdentifier = [[NSBundle mainBundle] bundleIdentifier];
     return @{
-        @"verifier": verifier,
-        @"code_challenge": [self sign:verifier],
-        @"code_challenge_method": @"S256",
-        @"state": [self randomValue]
-    };
+             @"verifier": verifier,
+             @"code_challenge": [self sign:verifier],
+             @"code_challenge_method": @"S256",
+             @"state": [self randomValue]
+             };
 }
 
 #pragma mark - SFSafariViewControllerDelegate
 
 - (void)safariViewControllerDidFinish:(SFSafariViewController *)controller {
     NSDictionary *error = @{
-        @"error": @"a0.session.user_cancelled",
-        @"error_description": @"User cancelled the Auth"
-    };
+                            @"error": @"a0.session.user_cancelled",
+                            @"error_description": @"User cancelled the Auth"
+                            };
     [self terminateWithError:error dismissing:NO animated:NO];
 }
 
 - (void)safariViewController:(SFSafariViewController *)controller didCompleteInitialLoad:(BOOL)didLoadSuccessfully {
-    if (!didLoadSuccessfully) {
+    if (self.didLoadCallback) {
+        didLoadSuccessfully ? self.didLoadCallback(@[[NSNull null]]) : self.didLoadCallback(@[@{@"error": @"failed to load"}]);
+        self.didLoadCallback = nil;
+    } else if (!didLoadSuccessfully) {
         NSDictionary *error = @{
-            @"error": @"a0.session.failed_load",
-            @"error_description": @"Failed to load authorize url"
-        };
+                                @"error": @"a0.session.failed_load",
+                                @"error_description": @"Failed to load authorize url"
+                                };
         [self terminateWithError:error dismissing:YES animated:YES];
```

Comment:
What are you thinking here, refactor it all to use the `didLoadCallback` and make the auto close optional?